tests/unit_tests/bittensor_tests/test_axon.py

```diff
@@ -1263,6 +1265,105 @@ def test_external_ip_port_set_full_address_internal(self):
         assert f'{internal_ip}:{internal_port}' == full_address1, f'{internal_ip}:{internal_port} is not eq to {full_address1}'
         assert f'{external_ip}:{external_port}' != full_address1, f'{external_ip}:{external_port} is eq to {full_address1}'
 
+class TestInterceptor(unittest.TestCase):
+
+    def setUp(self):
+        try:
+            os.remove('./nonces_dict.txt')
+        except:
+            Exception('No nonce_dict to be removed.')
```

Comment:
I know that we are not doing this in so many different tests but...I am going to share my point of view with this kind of practices in terms of unit testing.

In order to test this  piece of code it is pointed out as good practices to avoid to manage exceptions and we should not allow to have side effects generally, like the creation of the file by the code being tested.
In order to do it so, we should mock that piece of code so the file is not created, so we do not have to remove it. Tests should be run in a way that no side effects is produced

Reply:
The file was created from the [test_saving](https://github.com/opentensor/bittensor/pull/1253/files/68326b438dcecf317a9f1d1ec916e7245e6a3c61#diff-9bb2374a00ee2f68a526e31be4b02d7c663f17bad6e67b2693194fa7f4852a08R1287) and [test_loading] (https://github.com/opentensor/bittensor/pull/1253/files/68326b438dcecf317a9f1d1ec916e7245e6a3c61#diff-9bb2374a00ee2f68a526e31be4b02d7c663f17bad6e67b2693194fa7f4852a08R1296) function that is targeting at the [saving](https://github.com/opentensor/bittensor/pull/1253/files#diff-369f767fecee3a55495f39d9f75d496a6a61cac45c0ea2491760d190d38695d3R393) and [loading](https://github.com/opentensor/bittensor/pull/1253/files/68326b438dcecf317a9f1d1ec916e7245e6a3c61#diff-369f767fecee3a55495f39d9f75d496a6a61cac45c0ea2491760d190d38695d3R386) function here which are some basic file reading/writing function. So I am not sure how can we test it with mock, or is it actually not necessary to get these 2 function tested?

Reply:
yeah I know it's beign created. In terms of logic this is Ok. For the sake of simplicity and merge this, if you want, just proceed as it is, it's fine.

From the testing point of view, when testing that code that creates the file, we should mock that so no file is created. And we just intereact with the Mock. If you want just merge as it is and let's create a tech-deb task for that

Reply:
Thanks edu, i understand, let me change the test to mock before merging it:)